### PR TITLE
Remove jscs and jshint in favor of eslint

### PR DESCRIPTION
**Pull request by steveoh**
[eslint](http://eslint.org/) looks great. Its rules seem to be very comprehensive and jscs/jshint rules were all carried over. This allows us to simplify the number of tasks for grunt to run. [force](https://github.com/sindresorhus/grunt-eslint/pull/55) doesn't work but there is a module to use instead. eslint found a few things in my code that jscs should have found and didn't and a few other niceties that I enjoy having. 

I think this should end up in the boilerplate.

you'll want `apm install linter-eslint`

There is also an [html linter plugin](https://github.com/BenoitZugmeyer/eslint-plugin-html) for eslint that I think we should investigate.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{
```
Comment (stdavis) [inline .eslintrc]: Would be nice if this was 4 space indentation for consistency.

File: LayerSelector.js [hunk 2]
```diff
@@ -182,8 +182,8 @@ define([
                 return;
             }
 
-            var top = this.top != null ? this.top : true;
-            var right = this.right != null ? this.top : true;
+            var top = this.top != null ? this.top : true; // eslint-disable-line no-eq-null, eqeqeq
```
Comment (stdavis) [inline LayerSelector.js]: Why not `!==` and remove the need to disable the eslint error?

Comment (steveoh) [inline LayerSelector.js]: because it needs to check for null and undefined.

Comment (stdavis) [inline LayerSelector.js]: A better method may be `var top = this.top || true;`.

Comment (steveoh) [inline LayerSelector.js]: what if they have this.top set to false?

Comment (stdavis) [inline LayerSelector.js]: OK. Now I see. Seems like a messy situation but I'm good with it for now.

**Pull request merged**